proposals/NNNN-typed-throws-concurrency-library.md

```diff
@@ -0,0 +1,319 @@
+# Typed throws
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Doug Gregor](https://github.com/DougGregor), [Konrad Malawski](https://github.com/ktoso), [Franz Busch](https://github.com/FranzBusch)
+* Review Manager: TBD
+* Status: **Proposed**
+
+## Introduction
+
+SE-NNNN introduced typed error throws to Swift which allows developers to explicitly
+state the thrown errors.
+
+Swift-evolution threads:
+
+* [Typed throw functions - Evolution / Discussion - Swift Forums](https://forums.swift.org/t/typed-throw-functions/38860)
+* [Status check: typed throws](https://forums.swift.org/t/status-check-typed-throws/66637)
+
+
+## Motivation
+
+There are a number of places in the `Concurrency` library where the adoption of typed
+throws will help maintain thrown types through user code.
+
+
+## Proposed solution
+
+### `Task` creation and completion
+
+The [`Task`](https://developer.apple.com/documentation/swift/task) APIs have a
+`Failure` type similarly to `Result`, but use a pattern of overloading on
+`Failure == any Error` and `Failure == Never` to handling throwing and
+non-throwing versions. For example, the `Task` initializer is defined as the
+following overloaded pair:
+
+```swift
+init(priority: TaskPriority?, operation: () async -> Success) where Failure == Never
+init(priority: TaskPriority?, operation: () async throws -> Success) where Failure == any Error
+```
+
+These two initializers can be replaced with a single initializer using typed throws:
+
+```swift
+init(priority: TaskPriority?, operation: () async throws(Failure) -> Success)
```

Comment:
```suggestion
 @discardableResult
 init(priority: TaskPriority?, operation: () async throws(Failure) -> Success)
```

you're including the @discardableResult in signatures below, so it should also be here for completeness